src/pepr/prometheus/index.ts

```diff
@@ -35,10 +34,14 @@ When(PrometheusServiceMonitor)
       sm.Raw.metadata?.annotations?.["uds/skip-sm-mutate"] ||
       !(await isIstioInjected(sm))
     ) {
-      log.info(
-        `Mutating scrapeClass to exempt ServiceMonitor ${sm.Raw.metadata?.name} from default scrapeClass mTLS config`,
-      );
-      sm.Raw.spec.scrapeClass = "exempt";
+      // if scrapeClass is already set, don't override it
+      if (sm.Raw.spec.scrapeClass === undefined) {
```

Comment:
The placement of this means that we still mutate things _for_ Istio mTLS even if a scrape class is set (i.e. the `else` condition here doesn't take a pre-defined `scrapeClass` into account).

Real example of where this could be problematic - if the [metrics-server servicemonitor](https://github.com/defenseunicorns/uds-core/blob/main/src/metrics-server/chart/templates/service-monitor.yaml#L6-L7) set `scrapeClass: exempt` rather than the annotation for skipping, `endpoint.scheme` would still end up getting mutated.

Am I thinking about this right? Wondering if we should just add an initial check in both for `if (sm.Raw.spec.scrapeClass != undefined)` rather after/OR-ed with `if (sm.Raw.spec === undefined)`?

Reply:
sounds good will do